hw05_parallel_execution/run.go

```diff
@@ -2,14 +2,87 @@ package hw05parallelexecution
 
 import (
 	"errors"
+	"sync"
 )
 
 var ErrErrorsLimitExceeded = errors.New("errors limit exceeded")
 
 type Task func() error
 
-// Run starts tasks in n goroutines and stops its work when receiving m errors from tasks.
-func Run(tasks []Task, n, m int) error {
-	// Place your code here.
+// Run starts tasks in N goroutines and stops its work when receiving M errors from tasks.
+func Run(tasks []Task, n int, m int) error {
+	if m <= 0 {
+		return ErrErrorsLimitExceeded
+	}
+
+	mt := sync.Mutex{}
+	wg := &sync.WaitGroup{}
+	errs := int64(0)
+
+	// Создаем каналы
+	taskCh := make(chan Task)
+	errCh := make(chan struct{})
+
+	// Добавляем в группу кол-во корутин
+	wg.Add(n)
+
+	// Пихаем горутины-обработчики
+	for i := 0; i < n; i++ {
+		go worker(m, wg, errCh, taskCh, &errs, &mt)
+	}
+
+	// Синхрониризуем каналы и пихаем таски в канал
+	for _, task := range tasks {
+		select {
+		case <-errCh:
+		case taskCh <- task:
+		}
+	}
```

Comment:
```suggestion
	// Синхрониризуем каналы и пихаем таски в канал
	for _, task := range tasks {
		select {
		case <-errCh:
		case taskCh <- task:
		}
	}
```
taskCh := make(chan Task, n) 
for _, task = range, tasks {
taskCh <- task
}
Вот так тоже работает